deploy/crds/metal3.io_baremetalhosts_crd.yaml

```diff
@@ -203,6 +203,43 @@ spec:
             online:
               description: Should the server be online?
               type: boolean
+            rootDeviceHints:
+              description: Provide guidance about how to choose the device for the
+                image being provisioned.
+              properties:
+                deviceName:
+                  description: A Linux device name like "/dev/vda"
+                  type: string
+                hctl:
+                  description: A SCSI bus address like 0:0:0:0
+                  type: string
+                model:
+                  description: A vendor-specific device identifier
+                  type: string
+                rotational:
+                  description: Device rotational type
+                  type: boolean
+                serialNumber:
+                  description: Device serial number
+                  type: string
+                sizeGigabytes:
+                  description: Size of the device in Gigabytes
+                  type: integer
```

Comment:
In ironic this may include an operator, like ">= 4". I think it makes sense to support it here as well, wdyt?

Reply:
What is the default behavior if only a number is provided?

Reply:
Strict equality (yeah, not exactly convenient).

Reply:
Having 2 fields makes sense. That's more in line with what we've done in other similar APIs where we don't want to embed expressions in fields.

Reply:
It's not clear that we can support both min and max at the same time. I have renamed the field to `MinSizeGigabytes` and updated the behavior to always include `>=` in the value sent to ironic. If we can have an expression with both `>=` and `<=` then we can add a `MaxSizeGigabytes` field as well.

Reply:
This also raises the question of what operators we should be assuming for the other fields. And whether we need to allow variations.

Reply:
The latest version sets an explicit operator for each value we pass to ironic and does not allow the operators to be modified.